schema.go

```diff
@@ -71,3 +78,249 @@ func (schema DatabaseSchema) validateOperations(operations ...Operation) bool {
+	// type. Depending on its value, more information may be available in TypeObj.
+	// E.g: If Type == TypeEnum, TypeObj.Key.Enum contains the possible values
+	Type      ExtendedType
+	TypeObj   *ColumnTypeObject
+	Ephemeral bool
+	Mutable   bool
+}
+
+// ColumnTypeObject is a type object as per RFC7047
+type ColumnTypeObject struct {
+	Key         *BaseType
+	KeyRawMsg   *json.RawMessage `json:"key,omitempty"`
+	Value       *BaseType
+	ValueRawMsg *json.RawMessage `json:"value,omitempty"`
+	Min         int              `json:"min,omitempty"`
+	// Unlimited is expressed by the const value Unlimited (-1)
+	Max       int
+	MaxRawMsg *json.RawMessage `json:"max,omitempty"`
+}
+
+// BaseType is a base-type structure as per RFC7047
+type BaseType struct {
+	Type ExtendedType `json:"type"`
+	// Enum will be parsed manually and set to a slice
+	// of possible values. They must be type-asserted to the
+	// corret type depending on the Type field
+	Enum       OvsSet
+	EnumMsg    *json.RawMessage `json:"enum,omitempty"`
+	MinReal    float64          `json:"minReal,omitempty"`
+	MaxReal    float64          `json:"maxReal,omitempty"`
+	MinInteger int              `json:"minInteger,omitempty"`
+	MaxInteger int              `json:"maxInteger,omitempty"`
+	MinLength  int              `json:"minLength,omitempty"`
+	MaxLength  int              `json:"maxLength,omitempty"`
+	RefTable   string           `json:"refTable,omitempty"`
+	RefType    RefType          `json:"refType,omitempty"`
+}
+
+// TypeString returns a string representation of the (native) column type
+func (column *ColumnSchema) String() string {
+	var flags []string
+	var flagStr string
+	var typeStr string
+	if column.Ephemeral {
+		flags = append(flags, "E")
+	}
+	if column.Mutable {
+		flags = append(flags, "M")
+	}
+	if len(flags) > 0 {
+		flagStr = fmt.Sprintf("[%s]", strings.Join(flags, ","))
+	}
+
+	switch column.Type {
+	case TypeInteger, TypeReal, TypeBoolean, TypeString:
+		typeStr = string(column.Type)
+	case TypeUUID:
+		typeStr = fmt.Sprintf("uuid [%s (%s)]", column.TypeObj.Key.RefTable, column.TypeObj.Key.RefType)
+	case TypeEnum:
+		typeStr = fmt.Sprintf("enum (type: %s): %v", column.TypeObj.Key.Type, column.TypeObj.Key.Enum)
+	case TypeMap:
+		typeStr = fmt.Sprintf("[%s]%s", column.TypeObj.Key.Type, column.TypeObj.Value.Type)
+	case TypeSet:
+		var keyStr string
+		if column.TypeObj.Key.Type == TypeUUID {
+			keyStr = fmt.Sprintf(" [%s (%s)]", column.TypeObj.Key.RefTable, column.TypeObj.Key.RefType)
+		} else {
+			keyStr = string(column.TypeObj.Key.Type)
+		}
+		typeStr = fmt.Sprintf("[]%s (min: %d, max: %d)", keyStr, column.TypeObj.Min, column.TypeObj.Max)
+	default:
+		typeStr = fmt.Sprintf("Unknown Type")
+	}
+
+	return fmt.Sprintf(strings.Join([]string{typeStr, flagStr}, " "))
+}
+
+// UnmarshalJSON unmarshalls a json-formatted column
+func (column *ColumnSchema) UnmarshalJSON(data []byte) error {
+	var _col ColumnJSON
```

Comment:
underscore in the first char of var name?

Reply:
It was just to identify that variable as a throughout temporary variable used to unmarshal what we know how to unmarshal. I have no strong opinion on this, I'll rename it.